crates/oxc_linter/src/rules/jsx_a11y/no_static_element_interactions.rs

```diff
@@ -0,0 +1,273 @@
+    ///
+    /// ```jsx
+    /// <div
+    ///   onClick={onClickHandler}
+    ///   onKeyPress={onKeyPressHandler}
+    ///   tabindex="0">
+    ///   Save
+    /// </div>
+    /// ```
+    ///
+    /// ```jsx
+    /// <div onClick={onClickHandler}>Save</div>
+    /// ```
+    ///
+    /// ```jsx
+    /// <span onKeyDown={onKeyDownHandler} tabIndex={0}>Open</span>
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    ///
+    /// ```jsx
+    /// <div
+    ///   onClick={onClickHandler}
+    ///   onKeyPress={onKeyPressHandler}
+    ///   role="button"
+    ///   tabindex="0">
+    ///   Save
+    /// </div>
+    /// ```
+    ///
+    /// Provide an appropriate role:
+    ///
+    /// ```jsx
+    /// <div role="button" onClick={onClickHandler}>Save</div>
+    /// ```
+    ///
+    /// Prefer native controls when possible (best):
+    /// ```jsx
+    /// <button onClick={onClickHandler}>Save</button>
+    /// ```
+    ///
+    /// ### Options
+    ///
+    /// #### `allowExpressionValues`
+    ///
+    /// `{ type: boolean, default: true }`
+    ///
+    /// Dynamic role values are allowed when `allowExpressionValues` is true:
+    /// ```jsx
+    /// <div role={isLink ? "link" : "button"} onClick={onClickHandler}>Go</div>
+    /// ```
+    ///
+    /// #### `handlers`
+    ///
+    /// ```ts
+    /// {
+    ///     type: string[],
+    ///     default: [
+    ///         "onClick",
+    ///         "onMouseDown",
+    ///         "onMouseUp",
+    ///         "onKeyPress",
+    ///         "onKeyDown",
+    ///         "onKeyUp",
+    ///     ]
+    /// }
+    /// ```
+    ///
+    NoStaticElementInteractions,
+    jsx_a11y,
+    suspicious,
+);
+
+impl Rule for NoStaticElementInteractions {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::JSXOpeningElement(jsx_opening_el) = node.kind() else {
+            return;
+        };
+
+        let element_type = get_element_type(ctx, jsx_opening_el);
+        if !HTML_TAG.contains(element_type.as_ref()) {
+            return;
+        }
+
+        let has_interactive_props = self.0.handlers.iter().any(|prop| {
+            has_jsx_prop(jsx_opening_el, prop)
+                .is_some_and(|attr| get_prop_value(attr)
+                .is_some())
+        });
+
+        if !has_interactive_props
+            || is_hidden_from_screen_reader(ctx, jsx_opening_el)
+            || is_presentation_role(jsx_opening_el)
+        {
+            return;
+        }
+
+        if is_interactive_element(element_type.as_ref(), jsx_opening_el)
+            || is_interactive_role(jsx_opening_el)
+            || is_noninteractive_element(element_type.as_ref(), jsx_opening_el)
+            || is_noninteractive_role(element_type.as_ref(), jsx_opening_el)
+            || is_abstract_role(element_type.as_ref(), jsx_opening_el)
+        {
+            // This rule has no opinion about abstract roles.
+            return;
+        }
```

Comment:
This comment doesn't seem to match the actual behavior, perhaps it should be more like this?


```suggestion
        if is_abstract_role(element_type.as_ref(), jsx_opening_el)
        {
            // This rule has no opinion about abstract roles.
            return;
        }
        
        if is_interactive_element(element_type.as_ref(), jsx_opening_el)
            && is_interactive_role(jsx_opening_el)
            && is_noninteractive_element(element_type.as_ref(), jsx_opening_el)
            && is_noninteractive_role(element_type.as_ref(), jsx_opening_el) {
            // actual rule logic here ...
        }
```